pyhik/hikvision.py

```diff
@@ -132,6 +132,84 @@ def current_event_states(self):
+        return self.motion_detection
+
+    def get_motion_detection(self):
+        url = '%s/MotionDetection/1' % self.root_url
+        try:
+            response = self.hik_request.get(url)
+        except requests.exceptions.RequestException as err:
+            _LOGGING.error('Unable to fetch MotionDetection, error: %s', err)
+            self.motion_detection = None
+            return self.motion_detection
+
+        if response.status_code == requests.codes.unauthorized:
+            _LOGGING.error('Authentication failed')
+            self.motion_detection = None
+            return self.motion_detection
+
+        if response.status_code != requests.codes.ok:
+            # If we didn't receive 200, abort
+            _LOGGING.debug('Unable to fetch motion detection.')
+            self.motion_detection = None
+            return self.motion_detection
+
+        try:
+            tree = ET.fromstring(response.text)
+            nmsp = tree.tag.split('}')[0].strip('{')
+            self.namespace = nmsp if nmsp.startswith('http') else XML_NAMESPACE
+            ET.register_namespace("", self.namespace)
+            enabled = tree.find(self.element_query('enabled'))
+
+            if enabled is not None:
+                self._motion_detection_xml = tree
+            self.motion_detection = {'true': True, 'false': False}[enabled.text]
+            return self.motion_detection
+
+        except AttributeError as err:
+            _LOGGING.error('Entire response: %s', response.text)
+            _LOGGING.error('There was a problem: %s', err)
+            self.motion_detection = None
+            return self.motion_detection
+
+    def enable_motion_detection(self):
+        self._set_motion_detection(True)
+
+    def disable_motion_detection(self):
+        self._set_motion_detection(False)
+
+    def _set_motion_detection(self, enable):
+        url = '%s/MotionDetection/1' % self.root_url
+        current_state = self.get_motion_detection()
+        if enable == current_state
+            return
+
+        enabled = self._motion_detection_xml.find(self.element_query('enabled'))
+        if enabled is None:
+            _LOGGING.error("Couldn't find 'enabled' in the xml")
+            _LOGGING.error('XML: %s', ET.tostring(self._motion_detection_xml))
+            return
+
+        enabled.text = 'true' if enable else 'false'
+        xml = ET.tostring(self._motion_detection_xml)
+
+        try:
+            response = self.hik_request.put(url, data=xml)
+        except requests.exceptions.RequestException as err:
+            _LOGGING.error('Unable to set MotionDetection, error: %s', err)
+            return
+
+        if response.status_code == requests.codes.unauthorized:
+            _LOGGING.error('Authentication failed')
+            return
+
+        if response.status_code != requests.codes.ok:
+            # If we didn't receive 200, abort
+            _LOGGING.error('Unable to set motion detection: %s', response.text)
+
```

Comment:
If we get no errors from the camera we should update `self.motion_detection` with the enable value.

Reply:
Good catch! I'll add a few unit tests.